compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -561,26 +565,28 @@ fn check_for_bindings_named_same_as_variants(
                 BINDINGS_WITH_VARIANT_NAME,
                 p.hir_id,
                 p.span,
+                DelayDm(|| format!(
```

Comment:
Idle thought: maybe let's make this argument *always* `DelayDm` (or maybe *"just"* the closure?) That way you make it less likely to make the mistake? Not necessary for this PR (just a thought so we don't impl yet another complex internal lint.)

Reply:
That is an option. Maybe a trait like `IntoLintMessage` implemented for `F: FnOnce() -> String` and the `fluent` thing. We could also do the same with `Into<SubdiagnosticMessage>`, but that would also affect non-lints, so idk.